### ISSUE TITLE
Integrate into new `DataStore` and `DataType` into `Mapeo` class

### ISSUE THREAD

**Issue by gmaclennan** [digidem/mapeo-core-next#148]
The architecture changes in #139 mean that the way that the main class `Mapeo` is composed needs some adjustments.

First step here is to integrate the `observation` dataType so that we can have a working prototype for CRUD methods on observations. We can add other data types in a follow-up issue.

**Comment by gmaclennan** [comment]
I've been wondering whether the `Mapeo` class should handle the storage lifecycle, or if it should take storage instances as parameters. It needs both a sqlite database instance and a function that returns random-access-storage instances.

Using storage instances as parameters is the most flexible, but it requires a lot of boilerplate to use the class. I think the simplest API might be for the class to handle storage lifecycle, e.g. include a `close()` method that closes all storage. It would unrecoverable (e.g. no `open()`) so maybe call it `destroy()`? This would require a `storagePath` parameter, defining the folder where all storage (SQLite db and random-access-storage) would reside.

For tests it's often convenient to use in-memory storage, so I'm thinking of leaving the `storagePath` optional, and using in-memory storage if it is undefined. However it might be best-practice to always run tests with disk storage, to get as close to real-life scenarios as possible.

**Comment by achou11** [comment]
@gmaclennan 

> Using storage instances as parameters is the most flexible, but it requires a lot of boilerplate to use the class.

I don't have very strong thoughts on which approach is best suited, but I'm usually in favor of providing parameters for these kinds of things at the cost of some implementation effort. Guess the question is how much additional implementation effort it is.

In practice, I often find myself wishing that those kinds of options are exposed as opposed to baked-in. Especially since this can be used in various environments/platforms, providing that flexibility can be useful and allows any relevant adjustments/fixes to be done by the consumer as opposed to waiting on the module.

All that aside, your proposal about the `storagePath` param stuff seems reasonable to me!
